scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -309,154 +309,230 @@ For this example we will focus on identifying DE genes in these `Tumor_Myoblast`
+We will use the [`aggregateAcrossCells()`function from the `scuttle` package](https://rdrr.io/github/LTLA/scuttle/man/aggregateAcrossCells.html) to pseudo-bulk our dataset.
+This function takes as input an SCE object and the grouping assignments for each cell.
+The output will be an SCE object that contains only the pseudo-bulked counts for all genes across all specified groups, rather than across all cells. 
+We can then subset this SCE to just include our cell type of interest (tumor myoblasts) for input to the DE analysis. 
 
-# column names aren't automatically added, so let's add them in 
-colnames(pseudobulked_sce) <- glue::glue("{pseudobulked_sce$celltype_broad}_{pseudobulked_sce$sample}")
+We can pseudo-bulk using any grouping that we are interested in.
+For right now, we are interested in looking at gene expression across cell types, so we want to group the pseudo-bulked counts matrix by both cell type and original sample. 
+
+```{r pseudobulk sce}
+# first subset the coldata 
+# to only have the columns we care about for pseudo-bulking 
+pb_groups <- colData(rms_sce)[, c("celltype_broad", "sample")]
+
+# create a new SCE object that contains 
+# the pseudo-bulked counts across the provided groups 
+pb_sce <- scuttle::aggregateAcrossCells(rms_sce, 
+                                        id = pb_groups)
+
+# column names aren't automatically added to the pseudo-bulked sce, 
+# so let's add them in 
+colnames(pb_sce) <- glue::glue(
+  "{pb_sce$celltype_broad}_{pb_sce$sample}"
+)
 
-pseudobulked_sce
+pb_sce
 ```
 
 How does the new pseudo-bulked `SingleCellExperiment` look different? 
+How many columns does it have? 
+
+Let's take a look at what the `colData` looks like in the pseudo-bulked SCE object. 
 
-```{r}
-# only 37 columns in the counts assay 
-counts(pseudobulked_sce)[1:10, 1:10]
+```{r pseudobulk colData}
+# note the new addition of the column with number of cells per group 
+head(colData(pb_sce))
 ```
 
-Let's take a look at what the `colData` now looks like in the pseudo-bulked SCE object. 
+You should see that columns such as `sum`, `detected`, `subsets_mito_sum`, and other columns that typically contain per cell QC statistics now contain `NA` rather than numeric values. 
+This is because these values were initially calculated on a per cell level (we did this using `scuttle::addPerCellQCMetrics()`), but we no longer have a single column per cell, each column represents a group of cells, in this case cells of a specified cell type from an individual sample.
+Therefore, the numbers that we calculated on a per cell level are no longer applicable to this pseudo-bulked SCE object.
 
-```{r}
-# note the new addition of the column with number of cells per group 
-colData(pseudobulked_sce)
+You should also see a new column that wasn't present previously, the `ncells` column.
+This column was added during pseudo-bulking and indicates the total number of cells that were summed together to form each column of the SCE object.
```

Comment:
I moved it so that I first talk about the columns we've seen before, the QC columns and mention why they have NA. Then I talk about the new `ncells` column.